File: app/src/main/java/com/webviewblock/domain/interactors/GetSettingsUseCase.java
package com.webviewblock.domain.interactors;

import com.webviewblock.app.repository.LocalSettingsRepository;
import com.webviewblock.domain.History;

import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class GetSettingsUseCase {

    LocalSettingsRepository repository;

    @Inject
    public GetSettingsUseCase(LocalSettingsRepository repository) {
        this.repository = repository;
    }

    public List<History> execute() {
        return repository.getHistory();
    }

    public boolean isBlockImagesPreference() {
        return repository.isBlockImagesPreference();
    }

    public void setBlockImagesPreference(Boolean block) {
        repository.setBlockImagesPreference(block);
    }

    public void clear() {
        repository.clear();
    }

    public void updateHistory(List<History> historyList) {
        repository.updateHistory(historyList);
    }
}

File: app/src/main/java/com/webviewblock/domain/interactors/GetSettingsUseCase.kt
package com.webviewblock.domain.interactors

import com.webviewblock.app.repository.LocalSettingsRepository
import com.webviewblock.domain.History
import com.webviewblock.util.BehaviourSubjectDecorator
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class GetSettingsUseCase @Inject constructor(
    var repository: LocalSettingsRepository
) {

    fun execute(): BehaviourSubjectDecorator<List<History>> {
        return repository.history
    }

    fun blockImagesPreference(): Boolean {
        return repository.blockImagesPreference
    }

    fun setBlockImagesPreference(block: Boolean) {
        repository.blockImagesPreference = block
    }

    fun clear() {
        repository.clear()
    }
}
